Example/ViewController.swift

```diff
@@ -526,18 +526,24 @@ class ViewController: UIViewController {
     }
         
     fileprivate lazy var defaultSuccess: RouteRequestSuccess = { [weak self] (response) in
-        guard let routes = response.routes, !routes.isEmpty, case let .route(options) = response.options else { return }
-        self?.navigationMapView.removeWaypoints()
-        self?.response = response
+        guard let self = self,
+              // In some rare cases this callback can be called when `NavigationMapView` object is no
+              // longer available. This check prevents access to invalid object.
```

Comment:
Easiest way to reproduce this type of crash is to disable and then quickly enable it.